src/main/java/SuperMarket.java

```diff
@@ -1,19 +1,24 @@
+import java.math.BigDecimal;
 import java.util.ArrayList;
+import java.util.Collections;
 import java.util.List;
 
 public class SuperMarket {
     List<Inventory> inventoryList = new ArrayList<>();
+    List<Cart> myCart = new ArrayList<>();
     List<Promotions> promoList = new ArrayList<>();
     Double cartPrice = 0d;
     Double individualItemPrice = 0d;
     Double promoPrice = 0d;
+    int receiptNo = 1;
 
     public List<Inventory> initializeInventory() {
         return inventoryList;
     }
 
-    public void addItem(Inventory inventory) {
-        if (inventory.itemPrice > 0 && inventory.storeQty > 0) {
+    public void addItemToInventory(Inventory inventory) {
+        boolean hasSerialNo = inventory.serialNumber != "" && inventory.serialNumber != null;
```

Comment:
Bad condition.
Expressions are evaluated from left to right.
In this situation if inventory.serialNumber will be null NPE will be throwed because first condition is `inventory.serialNumber != ""`

Reply:
Changed to comparison to ||, please check

Reply:
`serialNumber != null && !serialNumber.equals("")` is needed there